pkg/cmd/pulumi/about.go

```diff
@@ -904,29 +918,41 @@ func getProjectRuntimeAbout(proj *workspace.Project) (projectRuntimeAbout, error
 			return projectRuntimeAbout{}, err
 		}
 		if out, err = cmd.Output(); err != nil {
-			return projectRuntimeAbout{}, fmt.Errorf("Failed to get python version: %w", err)
+			return errCouldNotGet(err)
 		}
 		version = "v" + strings.TrimPrefix(string(out), "Python ")
 	case langGo:
 		ex, err = executable.FindExecutable("go")
 		if err != nil {
-			return projectRuntimeAbout{}, fmt.Errorf("Could not find python executable: %w", err)
+			return errCouldNotFind(err)
 		}
 		cmd := exec.Command(ex, "version")
 		if out, err = cmd.Output(); err != nil {
-			return projectRuntimeAbout{}, fmt.Errorf("Failed to get go version: %w", err)
+			return errCouldNotGet(err)
 		}
 		version = "v" + strings.TrimPrefix(string(out), "go version go")
 	case langDotnet:
 		ex, err = executable.FindExecutable("dotnet")
 		if err != nil {
-			return projectRuntimeAbout{}, fmt.Errorf("Could not find dotnet executable: %w", err)
+			return errCouldNotFind(err)
 		}
 		cmd := exec.Command(ex, "--version")
 		if out, err = cmd.Output(); err != nil {
-			return projectRuntimeAbout{}, fmt.Errorf("Failed to get dotnet version: %w", err)
+			return errCouldNotGet(err)
 		}
 		version = "v" + string(out)
+	case langYAML:
+		return projectRuntimeAbout{Language: language}, nil
+	case langJava:
+		ex, err = executable.FindExecutable("javac")
```

Comment:
```java --version``` is useful also (JRE), javac is the JDK piece.

```mvn --version``` and ```gradle --version``` perhaps; getting a bit more verbose but may be useful. Note that those may be missing and that may be ok.